src/commands/createFunction/openAPISteps/OpenAPICreateStep.ts

```diff
@@ -0,0 +1,130 @@
+                break;
+            case ProjectLanguage.CSharp:
+                args.push(`--namespace:${nonNullProp(wizardContext, 'namespace')}`);
+                args.push('--azure-functions-csharp');
+                break;
+            case ProjectLanguage.Java:
+                args.push(`--namespace:${nonNullProp(wizardContext, 'javaPackageName')}`);
+                args.push('--azure-functions-java');
+                break;
+            case ProjectLanguage.Python:
+                args.push('--azure-functions-python');
+                args.push('--no-namespace-folders:True');
+                args.push('--no-async');
+                break;
+            default:
+                throw new Error(localize('notSupported', 'Not a supported language. We currently support C#, Java, Python, and Typescript'));
+                break;
+        }
+
+        args.push('--generate-metadata:false');
+        args.push(`--output-folder:${cpUtils.wrapArgInQuotes(wizardContext.projectPath)}`);
+
+        ext.outputChannel.appendLog(localize('statutoryWarning', 'Using the plugin could overwrite your custom changes to the functions.\nIf autorest fails, you can run the script on your command-line, or try resetting autorest (autorest --reset) and try again.'));
+        const title: string = localize('generatingFunctions', 'Generating from OpenAPI...Check [output window](command:{0}) for status.', ext.prefix + '.showOutputChannel');
+
+        await window.withProgress({ location: ProgressLocation.Notification, title }, async () => {
+            await cpUtils.executeCommand(ext.outputChannel, undefined, 'autorest', ...args);
+        });
+
+        if (wizardContext.language === ProjectLanguage.TypeScript) {
+            // Change the package.json only when the autorest successfully ran.
+            await addAutorestSpecificTypescriptDependencies(wizardContext);
+        }
+    }
+
+    public shouldExecute(): boolean {
+        return true;
+    }
+}
+
+async function validateAutorestInstalled(context: IActionContext): Promise<void> {
+    try {
+        await cpUtils.executeCommand(undefined, undefined, 'autorest', '--version');
+    } catch (error) {
+        const message: string = localize('autorestNotFound', 'Failed to find "autorest" | Extension needs AutoRest to generate a function app from an OpenAPI specification. Click "Learn more" for more details on installation steps.');
+        if (!context.errorHandling.suppressDisplay) {
+            window.showErrorMessage(message, DialogResponses.learnMore).then(async result => {
+                if (result === DialogResponses.learnMore) {
+                    await openUrl('https://aka.ms/autorest');
+                }
+            });
+            context.errorHandling.suppressDisplay = true;
+        }
+
+        throw new Error(message);
+    }
+}
+
+async function addAutorestSpecificTypescriptDependencies(context: IFunctionWizardContext): Promise<void> {
```

Comment:
A few problems with this code:
1. It uses `coreHttp` as the package name instead of `@azure/core-http` in package.json
1. It asks to overwrite `package.json` when creating a project in an empty folder
1. It always overwrites the entire file, instead of editing the existing file

Here are some better options without those problems.

### Option 1:
```typescript
await cpUtils.executeCommand(ext.outputChannel, context.projectPath, 'npm', 'install', '@azure/core-http', '--save-dev');
```

### Option 2:
```typescript
const coreHttp: string = '@azure/core-http';
const coreHttpVersion: string = '^1.1.4';
const packagePath: string = path.join(context.projectPath, 'package.json');

await confirmEditJsonFile(packagePath, (data: { devDependencies?: { [key: string]: string } }): {} => {
    // tslint:disable-next-line: strict-boolean-expressions
    data.devDependencies = data.devDependencies || {};
    if (!data.devDependencies[coreHttp]) {
        data.devDependencies[coreHttp] = coreHttpVersion;
    }
    return data;
});
```

Reply:
@EricJizbaMSFT Thanks. I've used the 2nd option to do this work.